modules/openapi-generator/src/main/resources/dart2/class.mustache

```diff
@@ -6,7 +6,12 @@ class {{classname}} {
   {{#min}} // range from {{min}} to {{max}}{{/min}}//{{^min}}enum {{name}}Enum { {{#values}} {{.}}, {{/values}} };{{/min}}{
   {{/allowableValues}}
   {{/vars}}
-  {{classname}}();
```

Comment:
@agilob I'm no expert in Dart but shall we keep the default constructor (no argument) for backward compatibility?

(upcoming 5.0.0 allows breaking changes without fallbacks but it's still worth considering backward compatibility)

Reply:
Sure, happy to explain this feature. Dart has flexible syntax on method params, default values, and mandatory params.

Method/constructor params can be mandatory (as not named) or named (optional) or named AND mandatory. This change introduces constructors that have named optional parameters. Such parameters have input params inside `{}` brackets. Since all fields in these classes are public anyway (dart discourages use of setters and getters "just because"  like in java), so we have this syntax, of all valid constructors.

Constructors with all optional params are backwards compatible, it's really just programmers' preference
![image](https://user-images.githubusercontent.com/2544251/85627649-bf434e00-b666-11ea-90e1-75b42a5f39c6.png)


This MR introduces optional named params in constructors, while cascade operators is already supported, it's fully backwards compatible.
https://dart.dev/guides/language/language-tour#optional-parameters
https://dart.dev/guides/language/language-tour#cascade-notation-

>(upcoming 5.0.0 allows breaking changes without fallbacks but it's still worth considering backward compatibility)

I am myself a guardian of backwards compatibility ;)

This MR 6729 breaks backwards compatibility when creating enums (which now are just string without any validation), so I found it necessary.

Reply:
> I am myself a guardian of backwards compatibility ;)

👍 

Thanks for the explanation. The change looks good to me. 

If no further feedback or question, I'll merge it over the weekend.